rollup/main/default/classes/Rollup.cls

```diff
@@ -1534,11 +1557,40 @@ global without sharing virtual class Rollup implements Database.Batchable<SObjec
     return matchingMetadata;
   }
 
+  private static QueryWrapper getIntermediateGrandparentQueryWrapper(String grandparentFieldPath, List<SObject> calcItems, Map<Id, SObject> oldCalcItems) {
+    if(isCDC) {
+      return new QueryWrapper();
+    }
+
+    SObjectType sObjectType = calcItems[0].getSObjectType();
+    SObjectField fieldToken = getPartOfGrandparentChain(grandparentFieldPath, sObjectType);
+    String relationshipName = fieldToken.getDescribe().getRelationshipName();
+    Integer relationshipIndex = grandparentFieldPath.indexOf(relationshipName) + relationshipName.length();
+    String priorFieldPath = grandparentFieldPath.substring(0, relationshipIndex) + '.Id';
+    QueryWrapper wrapper = new QueryWrapper('', priorFieldPath);
+
+    for (SObject calcItem : calcItems) {
+      SObject oldCalcItem = oldCalcItems.containsKey(calcItem.Id) ? oldCalcItems.get(calcItem.Id) : calcItem;
+      String currentLookup = (String) calcItem.get(fieldToken);
+      String oldLookup = (String) oldCalcItem.get(fieldToken);
+      if (currentLookup != oldLookup) {
+        wrapper.addRecordId(currentLookup);
+        wrapper.addRecordId(oldLookup);
+      }
+    }
+    return wrapper;
+  }
+
```

Comment:
It's only because you convinced me months ago to use apex-prettier that I feel obligated to point out... you have a few extra line breaks (line 1583 and 1587), you might want to run prettier on this class

Reply:
Damn! I just saw this comment after pushing a small cosmetic change and apex prettier did run (I’ve taken to trying to only run it right before committing). I think it bombs out some times though, honestly; it didn’t seem to pick up those extra line breaks. I will fix!